armi/materials/uranium.py

```diff
@@ -20,35 +20,280 @@
 .. [AAAFuels]  Kim, Y S, and Hofman, G L. AAA fuels handbook.. United States: N. p., 2003. Web. doi:10.2172/822554. .
 """
 
-from armi.materials.material import Material
+from numpy import interp
+
+from armi.materials.material import FuelMaterial
+from armi.nucDirectory import nuclideBases as nb
 from armi.utils.units import getTk
 
 
-class Uranium(Material):
+class Uranium(FuelMaterial):
     name = "Uranium"
 
+    enrichedNuclide = "U235"
+
     materialIntro = ""
 
     propertyNotes = {"thermal conductivity": ""}
 
     propertyRawData = {"thermal conductivity": ""}
 
-    propertyUnits = {"thermal conductivity": "W/m-K"}
+    propertyUnits = {"thermal conductivity": "W/m-K", "heat capacity": "J/kg-K"}
 
     propertyEquation = {
         "thermal conductivity": "21.73 + 0.01591T + 5.907&#215;10<super>-6</super>T<super>2</super>"
     }
 
-    propertyValidTemperature = {"thermal conductivity": ((255.4, 1173.2), "K")}
+    _heatCapacityTableK = [
+        298,
+        300,
+        400,
+        500,
+        600,
+        700,
+        800,
+        900,
+        941.9,
+        942,
+        1000,
+        1048.9,
+        1049,
+        1100,
+        1200,
+        1300,
+        1400,
+        1407.9,
+        1408,
+        1500,
+        1600,
+        1700,
+        1800,
+        1900,
+        2000,
+        2100,
+        2200,
+        2400,
+    ]
+
+    _heatCapacityTable = [
+        27.665,
+        27.700,
+        29.684,
+        31.997,
+        34.762,
+        38.021,
+        41.791,
+        46.081,
+        48.038,
+        42.928,
+        42.928,
+        42.928,
+        38.284,
+        38.284,
+        38.284,
+        38.284,
+        38.284,
+        38.284,
+        48.660,
+        48.660,
+        48.660,
+        48.660,
+        48.660,
+        48.660,
+        48.660,
+        48.660,
+        48.660,
+        48.660,
+    ]  # J/K/mol
+
+    _densityTableK = [
+        293,
+        400,
+        500,
+        600,
+        700,
+        800,
+        900,
+        940.9,
+        941,
+        1000,
+        1047.9,
+        1048,
+        1100,
+        1200,
+        1400,
+        1407.9,
+        1408,
+        1500,
+        1600,
+    ]
+
+    _densityTable = [
+        19.07,
+        18.98,
+        18.89,
+        18.79,
+        18.68,
+        18.55,
+        18.41,
+        18.39,
+        18.16,
+        18.11,
+        18.07,
+        17.94,
+        17.88,
+        17.76,
+        17.53,
+        17.52,
+        16.95,
+        16.84,
+        16.71,
+    ]  # g/cc
+
+    _linearExpansionPercent = [
```

Comment:
1. Can we get references here?
2. Are these references fully open-sourcable?

Reply:
The references were added to the class `references` dict:
https://github.com/terrapower/armi/pull/1247/files#diff-892fc46b16c9e9253d76ae4229b186fcde69674d805f3e3db5ca27ced4b32c4fR205-R216

Let me know if that's not sufficient.

All of them come from open documents, from Argonne.

Reply:
Oh, no, that's exactly what I wanted.  I am trying out the new GH code preview and I guess the slight change in UI got me.

Thanks!